app/src/features/Mappings/Create/CreateMapping.tsx

```diff
@@ -218,6 +223,14 @@ const CreateMapping = (): JSX.Element => {
     history.goBack();
   };
 
+  useEffect(() => {
+    dispatch(
+      resourceAdded({
+        id: uuid(),
+        source: sourceId,
+      })
+    );
+  }, []);
```

Comment:
Nice one, I got tired of having to go back and forth when refreshing the page ahah

Reply:
I guess we'll need this line too : 

```suggestion
  useEffect(() => {
    dispatch(
      resourceAdded({
        id: uuid(),
        source: sourceId,
      })
    );
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```

Reply:
Or, this should still work if you add `sourceId` & `dispatch` in the dependencies right ?